### PR TITLE
refactor: cleanup and optimize

### PR DESCRIPTION
**Pull request by abbotto**
### Changelog:
- Use `dumb-init` for Docker as `PID 1`
- Add `.gitignore` file
- Add `docker-compose.yml` file
- Add `.travis.yml` file
- Update the documentation
- Add `lint.sh` script
- Code refactoring
  - Compliant with `shellcheck` rules

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,21 +1,37 @@
-ARG pg_alpine_branch
```
Comment (dvalentiate) [inline Dockerfile]: these were deliberately lowercase to distinguish them from env vars.

**Review by dvalentiate — COMMENTED** [review]
Devs and ops will using AWS cli for lots of things. Not just s3 dumps. The idea isn't to just generate them inside the container and only be able to use them inside this container. The idea is to generate them and pass them to the container. If it is necessary to mount a volume it would be better to mount from the host into the contain for credentials than the other way around. Unless that is the purpose of the container. s3-dumping does not have the purpose of generating mfa credentials.

What file or data is being stored in the .aws directories that is being used. And can we can just that into the container and leave generating the credentials outside of the database dump container.

**Comment by dvalentiate** [comment]
Why are all the local variables in the .sh file in the .sh files getting uppercased? This doesn't follow [best practices](https://stackoverflow.com/questions/673055/correct-bash-and-shell-script-variable-capitalization). Constant and shell variables (export) get uppercase but everything else should be left alone.

**Comment by abbotto** [comment]
> Devs and ops will using AWS cli for lots of things. Not just s3 dumps. The idea isn't to just generate them inside the container and only be able to use them inside this container. The idea is to generate them and pass them to the container. If it is necessary to mount a volume it would be better to mount from the host into the contain for credentials than the other way around. Unless that is the purpose of the container. s3-dumping does not have the purpose of generating mfa credentials.
> 
> What file or data is being stored in the .aws directories that is being used. And can we can just that into the container and leave generating the credentials outside of the database dump container.

When I started this task it was my understanding that the goal was to get MFA working with S3 dumps for devs. I've recently made some changes to the code and and created a new repo for the AWS MFA session functionality.
- https://github.com/CoursePark/aws-mfa-session

File: Dockerfile (hunk 1 above)
Comment (abbotto) [reply Dockerfile]: 7524261

**Comment by abbotto** [comment]
> Why are all the local variables in the .sh file in the .sh files getting uppercased? This doesn't follow [best practices](https://stackoverflow.com/questions/673055/correct-bash-and-shell-script-variable-capitalization). Constant and shell variables (export) get uppercase but everything else should be left alone.

5721575

File: build_push.sh [hunk 2]
```diff
@@ -7,14 +7,15 @@ builds=$(echo '
...
 for build in $builds; do
-  tag=$(echo $build | cut -d ' ' -f 1 )
-  pgVersion=$(echo $build | cut -d ' ' -f 2)
-  pgAlpineBranch=$(echo $build | cut -d ' ' -f 3)
-  
-  echo docker build --tag bluedrop360/postgres-restore-from-s3:$tag --build-arg pg_version=$pgVersion --build-arg pg_alpine_branch=$pgAlpineBranch .
-  eval docker build --tag bluedrop360/postgres-restore-from-s3:$tag --build-arg pg_version=$pgVersion --build-arg pg_alpine_branch=$pgAlpineBranch .
-  echo docker push bluedrop360/postgres-restore-from-s3:$tag
-  eval docker push bluedrop360/postgres-restore-from-s3:$tag
+  TAG=$(echo "${build}" | cut -d ' ' -f 1 )
```
Comment (dvalentiate) [inline build_push.sh]: `TAG` should be replaced with `tag`.

Comment (abbotto) [reply build_push.sh]: e506897

File: action.sh [hunk 3]
```diff
@@ -1,72 +1,116 @@
-#!/bin/sh
+#!/usr/bin/env sh
```
Comment (dvalentiate) [inline action.sh:1]: this is pretty normal to do but really doesn't offer and advantages in a stripped down container

**Comment by dvalentiate** [comment]
bunch of things changing. Most of it seems good. Generally an improvement. Passes my code review. Haven't been able to test it.

**Comment by dvalentiate** [comment]
I approve. Please squash the changes during merge.

**Comment by dvalentiate** [comment]
Dumb-init is a good touch, shellcheck is probably a good idea but I really was just trying to keep this thing tiny without any bloat. But I guess it needs to grow up, so good idea. Same thought for travis and docker-compose. Looks like there is some improved messaging, Good move. Renaming every variable is really annoying.

**Review by dvalentiate — APPROVED** [review]

**Comment by abbotto** [comment]
> Renaming every variable is really annoying.

Initially, some variables were in `camel case` format and other variables were in `snake case` format.
I renamed the `camel case` variables to `snake case` format because that is what we use in other scripts.

**Pull request merged**